src/coreComponents/common/format/table/TableFormatter.cpp

```diff
@@ -96,315 +134,512 @@ string buildCell( TableLayout::Alignment const alignment, string_view value, int
   }
 }
 
-/**
- * @brief Detect columns who are not displayed from TableLayout and therefore modify columns / tableDataRows vectors
- * @param columns Vector built in TableLayout containing all columns with their parameters
- * @param tableDataRows Vector built in TableData containing all rows values
- */
-void formatColumnsFromLayout( std::vector< TableLayout::Column > & columns,
-                              std::vector< std::vector< string > > & tableDataRows )
-{
-  integer idxColumn = 0;
-  for( auto iterColumn = columns.begin(); iterColumn!=columns.end(); )
-  {
-    if( !iterColumn->m_parameter.enabled )
-    {
-      iterColumn = columns.erase( iterColumn );
-      for( auto & row : tableDataRows )
-      {
-        row.erase( row.begin() + idxColumn );
-      }
-    }
-    else
-    {
-      ++iterColumn;
-      ++idxColumn;
-    }
-  }
-}
-
 TableTextFormatter::TableTextFormatter( TableLayout const & tableLayout ):
   TableFormatter( tableLayout )
 {}
 
-void TableTextFormatter::fillTableColumnsFromRows( std::vector< TableLayout::Column > & columns,
-                                                   std::vector< std::vector< string > > & rows ) const
-{
-  for( size_t idxRow = 0; idxRow < rows.size(); idxRow++ )
-  {
-    if( rows[idxRow].size() < columns.size() )
-    {
-      rows[idxRow].resize( columns.size(), " " );
-    }
-
-    for( size_t idxColumn = 0; idxColumn < columns.size(); idxColumn++ )
-    {
-      if( m_tableLayout.getColumns()[idxColumn].m_parameter.enabled )
-      {
-        columns[idxColumn].m_columnValues.push_back( rows[idxRow][idxColumn] );
-      }
-    }
-  }
-}
-
-string TableTextFormatter::layoutToString() const
+string TableTextFormatter::toString() const
 {
   std::ostringstream tableOutput;
-  string sectionSeparatingLine;
-  std::vector< TableLayout::Column > columns = m_tableLayout.getColumns();
-
-  outputLayout( tableOutput, columns, {}, sectionSeparatingLine );
+  TableData tableData;
+  CellLayoutRows cellsHeaderLayout;
+  CellLayoutRows cellsDataLayout;
+  string separatorLine;
+
+  TableLayout tableLayout = m_tableLayout;
+  initalizeTableLayout( tableLayout, tableData, cellsHeaderLayout, cellsDataLayout, separatorLine );
+  outputTable( tableLayout, tableOutput, cellsHeaderLayout, cellsDataLayout, separatorLine );
   return tableOutput.str();
 }
```

Comment:
Would it do the job?
```suggestion
string TableTextFormatter::toString() const
{
  TableData tableData;
  return toString( tableData );
}
```

Reply:
indeed